refactor(feature:profile) : move all hardcoded strings to string resources

Nithish-Sri-Ram opened the pull request:
## Issue Fix
Fixes #1888
Jira Task: MW-237 https://mifosforge.jira.com/browse/MW-237

## Screenshots
<img width="360" height="740" alt="Screenshot_20250720_233924" src="https://github.com/user-attachments/assets/7fa745a7-a5d9-4a9d-8468-f45ea4aac4c4" />


## Description
This PR moves all hardcoded strings in the feature/profile module into string resources :

EditProfileViewModel.kt, to use string resources for validation error messages
ProfileScreen.kt to properly resolve StringResource objects in dialog components

The ci-prepush.sh - the build was successful, but as last time the ./gradlew check did give me error and didn't build pointing to the files which I've not made any changes to. 

<img width="1785" height="668" alt="Screenshot 2025-07-21 004846" src="https://github.com/user-attachments/assets/cb04509b-b007-48fb-8798-ccc109f7e1d0" />

<img width="1806" height="1046" alt="Screenshot 2025-07-20 205735" src="https://github.com/user-attachments/assets/e0a6b2d9-bd99-4b9f-817f-0d01cdaf4397" />

##
<!--Please make sure these boxes are checked before submitting your pull request - thanks!-->

- [ ] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [ ] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

Nithish-Sri-Ram committed:
refactor(feature:profile) : move all hardcoded strings to string resources

revanthkumarJ commented on feature/profile/src/commonMain/kotlin/org/mifospay/feature/profile/edit/EditProfileScreen.kt:
```diff
@@ -194,21 +196,35 @@ private fun EditProfileScreenContent(
     when (dialogState) {
-        is EditProfileState.DialogState.Error -> MifosBasicDialog(
+        is EditProfileDialog.Error -> { // Change this line
+            val message = if (dialogState.formatArgs.isNotEmpty()) {
```
can you remove this comment "// Change this line"

revanthkumarJ commented:
@Nithish-Sri-Ram if in your pr it shows like
<img width="1167" height="255" alt="image" src="https://github.com/user-attachments/assets/a84f38b7-3cd8-4dcb-b0d3-de4e29d423f2" />
then it means all checks are passed thats it.

Let me know once you remove that comment i will approve

Nithish-Sri-Ram committed:
Update EditProfileScreen.kt

niyajali committed:
Merge branch 'development' into MW-237

niyajali commented on feature/profile/src/commonMain/kotlin/org/mifospay/feature/profile/ProfileScreen.kt:
```diff
@@ -179,7 +180,7 @@ private fun ProfileScreenContent(
             },
             leadingIcon = {
-                Icon(imageVector = MifosIcons.AttachMoney, contentDescription = "")
+                Icon(imageVector = MifosIcons.AttachMoney, contentDescription = stringResource(Res.string.feature_profile_link_bank_account))
```
Format this line

niyajali commented on feature/profile/src/commonMain/kotlin/org/mifospay/feature/profile/edit/EditProfileViewModel.kt:
```diff
@@ -237,34 +235,46 @@ internal class EditProfileViewModel(
+
+            is DataState.Error -> {
+                mutableStateFlow.update {
+                    it.copy(dialogState = EditProfileDialog.ApiError(action.result.exception.message.toString()))
```
Format this

niyajali commented on feature/profile/src/commonMain/kotlin/org/mifospay/feature/profile/edit/EditProfileViewModel.kt:
```diff
@@ -237,34 +235,46 @@ internal class EditProfileViewModel(
     private fun handleUpdateClientImageResult(action: HandleUpdateClientImageResult) {
         when (action.result) {
+            is DataState.Success -> {
+                sendEvent(EditProfileEvent.ShowToast(Res.string.feature_profile_profile_image_updated_successfully))
```
Format this

niyajali commented on feature/profile/src/commonMain/kotlin/org/mifospay/feature/profile/edit/EditProfileViewModel.kt:
```diff
@@ -237,34 +235,46 @@ internal class EditProfileViewModel(
             is DataState.Error -> {
                 mutableStateFlow.update {
-                    it.copy(dialogState = Error(action.result.exception.message ?: ""))
+                    it.copy(dialogState = EditProfileDialog.ApiError(action.result.exception.message.toString()))
```
Format this

niyajali commented on feature/profile/src/commonMain/kotlin/org/mifospay/feature/profile/edit/EditProfileViewModel.kt:
```diff
@@ -237,34 +235,46 @@ internal class EditProfileViewModel(
                         is DataState.Success -> {
-                            sendEvent(EditProfileEvent.ShowToast("Profile updated successfully"))
-                            trySendAction(EditProfileAction.NavigateBack)
+                            sendEvent(EditProfileEvent.ShowToast(Res.string.feature_profile_profile_updated_successfully))
```
Format this lines properly

Nithish-Sri-Ram replied:
For the errors, I've followed the format Mr.Biplab has suggested, but I still don't understand what should be done differently for ShowToast?. Should I also create a sealed nested interface for the ShowToast like I did with the Error? @niyajali 

    sealed interface DialogState {
        data object Loading : DialogState
    
        sealed interface Error : DialogState {
            data class StringMessage(val message: String) : Error
            data class ResourceMessage(val message: StringResource) : Error
            // For dynamic messages with arguments
            data class GenericResourceMessage(val message: StringResource, val args: List<Any>) : Error
        }
    }

niyajali replied:
What's the issue with existing approach, aren't you're able show the toast message inside of EventsEffect handler.

And Yes, if required it could be done, for example of you're unable to use stringResources inside for showing the message, you could move that event to dialog state by adding one more error data class and by passing the snackbar state to the DialogComponent and you can access the stringResource and show the message on snackbar.

Nithish-Sri-Ram commented:
Understood, I've reverted back all the changes I've made and I've used the standard way of implementing everything, running the prepush.sh script, will be raising a pr soon for review @biplab1 and @niyajali. If any further changes are required, please do let me know. I've changed everything to use the implementation like this.

    sealed interface DialogState {
        data object Loading : DialogState
    
        sealed interface Error : DialogState {
            data class StringMessage(val message: String) : Error
            data class ResourceMessage(val message: StringResource) : Error
            // For dynamic messages with arguments
            data class GenericResourceMessage(val message: StringResource, val args: List<Any>) : Error
        }
    }

Please review and let me know if the below way of doing things is correct? I've made all the mentioned changes, will be raising a PR soon.


    is DataState.Error -> {
                    mutableStateFlow.update {
                        it.copy(dialogState = Error.StringMessage(action.result.exception.message ?: ""))
                    }
                }

Nithish-Sri-Ram committed:
Made all the suggested changes

Nithish-Sri-Ram committed:
Merge branch 'openMF:development' into MW-237

niyajali committed:
Merge branch 'development' into MW-237
